examples/topics/topicFilters.ts

```diff
@@ -0,0 +1,119 @@
+import { Namespace, SubscriptionClient, generateUuid } from "../../lib";
+import * as dotenv from "dotenv";
+import { CorrelationFilter } from '../../lib/core/managementClient';
+dotenv.config();
+
+const str = process.env.SERVICEBUS_CONNECTION_STRING || "";
+const topic = process.env.TOPIC_NAME || "";
+const subscriptionDefaultFilter = process.env.SUBSCRIPTION_NAME_DEFAULT_FILTER || "";
+const subscriptionSqlFilter = process.env.SUBSCRIPTION_NAME_SQL_FILTER || "";
+const subscriptionCorrelationFilter = process.env.SUBSCRIPTION_NAME_CORRELATION_FILTER || "";
+
+let ns: Namespace;
+
+const yellowSqlRule = {
+  name: "YellowSqlRule",
+  expression: "Color = 'Yellow'"
+};
+const correlationRule = {
+  name: "CorrelationRule",
+  filter: { label: "Blue Message" }
+};
+
+const redMessage = {
+  messageId: "RedMessageId" + generateUuid(),
+  body: "Red Message",
+  label: "Red Message",
+  userProperties: {
+    "Color": "Red"
+  }
+};
+
+const blueMessage = {
+  messageId: "BlueMessageId" + generateUuid(),
+  body: "Blue Message",
+  label: "Blue Message",
+  userProperties: {
+    "Color": "Blue"
+  }
+};
+
+const yellowMessage = {
+  messageId: "YelloMessageId" + generateUuid(),
+  body: "Yellow Message",
+  label: "Yellow Message",
+  userProperties: {
+    "Color": "Yellow"
+  }
+};
+
+async function main(): Promise<void> {
+  ns = Namespace.createFromConnectionString(str);
+
+  const subscriptionClientNoFilter = ns.createSubscriptionClient(topic, subscriptionDefaultFilter);
+  await removeAllRules(subscriptionClientNoFilter);
+  await testAddedRule(subscriptionClientNoFilter, subscriptionClientNoFilter.defaultRuleName, true);
+
+  const subscriptionClientSqlFilter = ns.createSubscriptionClient(topic, subscriptionSqlFilter);
+  await removeAllRules(subscriptionClientSqlFilter);
+  await testAddedRule(subscriptionClientSqlFilter, yellowSqlRule.name, yellowSqlRule.expression);
```

Comment:
Couldn't find an example exercising the sqlRuleActionExpression parameter in the
```js
addRule(ruleName: string, filter: boolean | string | CorrelationFilter, sqlRuleActionExpression?: string)
```

third parameter in the testAddedRule() is an expression. If this is intended to be used as an expression then the example is doing something different. `testAddedRule()` is always adding a rule with ruleName and a filter.

I guess testAddedRule should also have a third optional parameter for sqlRuleExpression and it should have an example for the same.

Reply:
Updated the `testAddedRule()` to include action as a parameter. Our future work on samples will cover the actual sample that will use the action.